x-pack/test/plugin_api_integration/test_suites/event_log/public_api_integration.ts

```diff
@@ -18,137 +18,162 @@ export default function ({ getService }: FtrProviderContext) {
   const supertest = getService('supertest');
   const log = getService('log');
   const retry = getService('retry');
+  const spacesService = getService('spaces');
+  const esArchiver = getService('esArchiver');
 
   describe('Event Log public API', () => {
-    it('should allow querying for events by Saved Object', async () => {
-      const id = uuid.v4();
+    before(async () => {
+      await spacesService.create({
+        id: 'namespace-a',
+        name: 'Space A',
+        disabledFeatures: [],
+      });
+    });
 
-      const expectedEvents = [fakeEvent(id), fakeEvent(id)];
+    after(async () => {
+      await esArchiver.unload('empty_kibana');
+    });
 
-      await logTestEvent(id, expectedEvents[0]);
-      await logTestEvent(id, expectedEvents[1]);
+    for (const namespace of [undefined, 'namespace-a']) {
```

Comment:
nit.

Could we use some kind of const here instead of just `undefined`?

something like:
```
const DEFAULT_NAMESPACE = undefined;
for (const namespace of [DEFAULT_NAMESPACE, 'namespace-a']) {
   const namespaceName = namespace === DEFAULT_NAMESPACE ? 'default' : namespace;
 /// ...
```

There's a bunch of places in this test where we use `undefined` to refer to the default namespace and I think, for ease of maintenance for future developers, that using a const when undefined means the default NS will make life a little easier.

Reply:
hmm, I actually tried that, but felt like keeping the definition of `namespace` to `string | undefined` still makes sense, since that's what it looks like in the non-test code.  So then changing the constant values besides that to a defined constant looked really weird:

```ts
function urlPrefixFromNamespace(namespace: string | undefined): string {
  if (namespace === DEFAULT_NAMESPACE) return '';
  return `/s/${namespace}`;
}
```

Feels better to just leave it as `undefined` to me.  There were only 4 references to the constant value outside the type definition anyway.